### PR TITLE
Template DataEntry on value_type

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

**Comment by peterrum** [comment]
I guess in a follow-up PR one needs to tackle `MGDataEntry`.

**Review by kronbichler — APPROVED** [review]

**Comment by masterleinad** [comment]
```
81 - simplex/step-68.mpirun=4.debug (Failed)
```

File: include/deal.II/numerics/data_out_dof_data.templates.h [hunk 1]
```diff
@@ -1068,8 +1068,9 @@ namespace internal
 
 
 
-    template <int dim, int spacedim, typename VectorType>
-    DataEntry<dim, spacedim, VectorType>::DataEntry(
+    template <int dim, int spacedim, typename value_type>
```
Comment (bangerth) [inline include/deal.II/numerics/data_out_dof_data.templates.h:1071]: We use `value_type` as a local typedef, and `ValueType` as the name of a template argument. Alternatively, `ScalarType` would work.

**Review by bangerth — CHANGES_REQUESTED** [review]
Looks good in principle, but I would like to change the name of the template argument.

**Comment by bangerth** [comment]
Not sure what's going on with that assertion, but that shouldn't be too hard to find.

**Comment by peterrum** [comment]
Done!

**Review by bangerth — APPROVED** [review]
Looks good, thanks!

**Pull request merged**